server/src/main/java/io/crate/planner/statement/CopyFromPlan.java

```diff
@@ -345,6 +347,15 @@ private static void rewriteToCollectToUsePartitionValues(List<Reference> partiti
         }
     }
 
+    private boolean waitForCompletion(ExecutionPlan plan) {
+        try {
+            return ((FileUriCollectPhase) ((Collect) plan).collectPhase()).withClauseOptions()
+                    .getAsBoolean("wait_for_completion", true);
+        } catch (Exception e) {
+            return true;
+        }
```

Comment:
What exception can occur here? 
If it's about the a possible cast error, I suggest to protect this with an assertion instead and inline the code in the caller.

Reply:
@seut can you have another look please?

Yes the try/catch is used only for a possible class cast exception. 
`CopyFromPlannerTest` has tests that check that plan is instance of the right classes and `CopyToPlannerTest` assumes correct classes and uses casts.

So can we remove completely any try/catch/assertion or use assertion as you suggested?

Reply:
I think we should try to get rid of the casts altogether.

You can change the way some methods interact and how the code is structured a bit. For example you could call `bind` first to get access to the `BoundCopyFrom` and then call  `planCopyFromExecution` with that. This would allow you to access the `wait_for_completion` option directly.